src/libs/actions/PersonalDetails.js

```diff
@@ -173,6 +173,74 @@ function updateDisplayName(firstName, lastName) {
     Navigation.navigate(ROUTES.SETTINGS_PROFILE);
 }
 
+/**
+ * @param {String} legalFirstName
+ * @param {String} legalLastName
+ */
+function updateLegalName(legalFirstName, legalLastName) {
+    API.write('UpdateLegalName', {legalFirstName, legalLastName}, {
+        optimisticData: [{
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: ONYXKEYS.PRIVATE_PERSONAL_DETAILS,
+            value: {
+                legalFirstName,
+                legalLastName,
+            },
+        }],
+    });
+    Navigation.navigate(ROUTES.SETTINGS_PERSONAL_DETAILS);
+}
+
+/**
+ * @param {String} dob - date of birth
+ */
+function updateDateOfBirth(dob) {
+    API.write('UpdateDateOfBirth', {dob}, {
+        optimisticData: [{
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: ONYXKEYS.PRIVATE_PERSONAL_DETAILS,
+            value: {
+                dob,
+            },
+        }],
+    });
+    Navigation.navigate(ROUTES.SETTINGS_PERSONAL_DETAILS);
+}
+
+/**
+ * @param {String} street
+ * @param {String} street2
+ * @param {String} city
+ * @param {String} state
+ * @param {String} zip
+ * @param {String} country
+ */
+function updateAddress(street, street2, city, state, zip, country) {
+    API.write('UpdateHomeAddress', {
+        addressStreet: street,
+        addressStreet2: street2,
+        addressCity: city,
+        addressState: state,
+        addressZipCode: zip,
+        addressCountry: country,
+    }, {
+        optimisticData: [{
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: ONYXKEYS.PRIVATE_PERSONAL_DETAILS,
+            value: {
+                address: {
+                    street: `${street}\n${street2}`,
+                    city,
+                    state,
+                    zip,
+                    country,
+                },
+            },
```

Comment:
Is there no validation for these address fields? Like if we update it while we are offline and it is a bad address what happens here?